### PR TITLE
push new repo

### PR DESCRIPTION
**Pull request by yuzhiguo07**

### CODE REVIEW

**Review by hehuanma — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -20,7 +20,7 @@ This package requires:
 * Scipy version >= 0.15.0  (For Anaconda users, use `conda install scipy` instead of `pip install scipy` to upgrade your version.)
 * [RDkit](http://www.rdkit.org/docs/Install.html)
 * [Autograd](http:github.com/HIPS/autograd) (Just run `pip install autograd`)
-
+* smile (run `pip install smile`)
```
Comment (shengwa) [inline README.md]: Please use the format above with the proper link.

File: data/18p/18p_1a2_data.csv [hunk 2]
```diff
@@ -0,0 +1,14227 @@
...
+Cc1c2cccc(Sc3ccccc3)c2c4C[S+](c5ccccc5)c6cccc1c46,0.0
+CC1=NC(=Cc2ccc(cc2)[N+](=O)[O-])C(=O)N1c3cc(Cl)ccc3C,1.0
+CC1=NC(=C(O)C(=O)N1)CN2CCCCC2,0.0
+Cc1ccc(Oc2cnc3ccccc3n2)cc1,1.0
+CC(=O)N1CCCC2(CCN(Cc3nccs3)C2)C1,0.0
+Clc1cccc(c1)C(=O)NC(=S)Nc2ccc3oc(nc3c2)c4cccnc4,0.0
+Fc1ccc(cc1F)C2=Nc3cncnc3N(C4CC4)C2=O,1.0
+Cc1cc(C)c(NC(=O)NC(=O)c2ccc(F)cc2)c(C)n1,0.0
+CCSC(=N)Nc1ccc(cc1)C(F)(F)F,1.0
+COCCn1c2CC3CN(C(=O)c4ccccc4)C(Cc5ccccc5)(C3c2cc1C(=O)N6CCCC6)C(=O)OC,0.0
```
Comment (shengwa) [inline data/18p/18p_1a2_data.csv]: Please remove the entire folder.

File: data/18p/18p_pgp_data.csv [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+CC(C)CC1NC(=O)[C@H](Cc2ccccc2)NC(=O)CNC(=O)[C@@H](C)NC(=O)[C@H](Cc3ccc(O)cc3)NC(=O)\C=C/c4ccccc4OC1=O,1.0
+Cc1ccsc1C(=CCCN2CCC[C@H](C2)C(=O)O)c3sccc3C,0.0
+CN(C)C(=O)Oc1ccc[n+](C)c1,0.0
+CCOC(=O)c1ncn2c1CN(C)C(=O)c3cc(F)ccc23,0.0
+COC(=O)C1=C(C)NC(=C(C1c2ccccc2[N+](=O)[O-])C(=O)OC)C,0.0
+COc1ccc2C[C@@H]3[C@@H]4CCC(=O)[C@@H]5Oc1c2[C@]45CCN3C,0.0
+C[C@H](Cc1ccccc1)N(C)CC#C,0.0
+CC[C@H]1OC(=O)[C@H](C)[C@@H](O[C@H]2C[C@@](C)(OC)[C@@H](O)[C@H](C)O2)[C@H](C)[C@@H](O[C@@H]3O[C@H](C)C[C@@H]([C@H]3O)N(C)C)[C@@](C)(C[C@@H](C)C(=O)[C@H](C)[C@@H](O)[C@]1(C)O)OC,1.0
+CC(=O)N1CCN(CC1)c2ccc(OC[C@H]3CO[C@@](Cn4ccnc4)(O3)c5ccc(Cl)cc5Cl)cc2,0.0
+CNCCCN1c2ccccc2CCc3ccccc13,0.0
```
Comment (shengwa) [inline data/18p/18p_pgp_data.csv]: Why do we need this in our code?

File: reg_logp.txt [hunk 4]
```diff
@@ -0,0 +1,53 @@
+Loading data...
+Task params {'target_name': 'logp', 'data_file': 'data/logp.csv'}
+
+Starting Morgan fingerprint experiment...
```
Comment (shengwa) [inline reg_logp.txt]: Are you sure you would like to add log file into the code base?

**Review by shengwa — COMMENTED** [review]
Please remove all the unrelated files and the code you haven't changed.

**Comment by shengwa** [comment]
Please check the PR yourself before requesting a code review.

File: neuralfingerprint/io_utils.py [hunk 5]
```diff
@@ -3,36 +3,44 @@
...
-    stops = [s.stop for s in slices]
-    if not all(stops):
-        raise Exception("Slices can't be open-ended")
+    return list(map(np.array, data))
 
-    data = read_csv(filename, max(stops), input_name, target_name)
-    return [(data[0][s], data[1][s]) for s in slices]
+
+def load_data(filename, input_name, target_name):
+    # slices = []
```
Comment (shengwa) [inline neuralfingerprint/io_utils.py]: Remove if no use.

File: neuralfingerprint/io_utils.py [hunk 6]
```diff
@@ -3,36 +3,44 @@
...
+    # start = 0
+    # for size in sizes:
+    #     stop = start + size
+    #     slices.append(slice(start, stop))
+    #     start = stop
+    return load_data_slices_nolist(filename, input_name, target_name)
+
+
+def load_data_slices_nolist(filename, input_name, target_name):
+    # stops = [s.stop for s in slices]
```
Comment (shengwa) [inline neuralfingerprint/io_utils.py]: Remove if no use.

File: neuralfingerprint/mol_graph.py [hunk 7]
```diff
@@ -62,8 +67,15 @@ def add_neighbors(self, neighbor_list):
     def get_neighbors(self, ntype):
         return [n for n in self._neighbors if n.ntype == ntype]
 
+
 def graph_from_smiles_tuple(smiles_tuple):
+    graph_list = []
```
Comment (shengwa) [inline neuralfingerprint/mol_graph.py]: Delete.

File: neuralfingerprint/mol_graph.py [hunk 8]
```diff
@@ -62,8 +67,15 @@ def add_neighbors(self, neighbor_list):
...
     graph_list = [graph_from_smiles(s) for s in smiles_tuple]
+    # for s in smiles_tuple:
+    #     if graph_from_smiles(s) is not None:
+    #         graph_list.append(graph_from_smiles(s))
```
Comment (shengwa) [inline neuralfingerprint/mol_graph.py]: Delete if no use.

File: neuralfingerprint/mol_graph.py [hunk 9]
```diff
@@ -72,22 +84,25 @@ def graph_from_smiles_tuple(smiles_tuple):
     big_graph.sort_nodes_by_degree('atom')
     return big_graph
 
+
 def graph_from_smiles(smiles):
     graph = MolGraph()
     mol = MolFromSmiles(smiles)
     if not mol:
+        # return None
```
Comment (shengwa) [inline neuralfingerprint/mol_graph.py]: Remove if no use.

File: neuralfingerprint/util.py [hunk 10]
```diff
@@ -64,26 +70,33 @@ def __call__(self, *args):
...
         return partial(self.__call__, obj)
 
+
 def normalize_array(A):
     mean, std = np.mean(A), np.std(A)
     A_normed = (A - mean) / std
+
     def restore_function(X):
-        return X * std + mean
+        # return X * std + mean
```
Comment (shengwa) [inline neuralfingerprint/util.py]: Add comments to keep this.

File: regression_logp.py [hunk 11]
```diff
@@ -0,0 +1,168 @@
...
+from neuralfingerprint import load_data
+from neuralfingerprint import build_morgan_deep_net
+from neuralfingerprint import build_conv_deep_net
+from neuralfingerprint import normalize_array, adam
+from neuralfingerprint import build_batched_grad
+from neuralfingerprint.util import rmse
+
+from autograd import grad
+
+flags.DEFINE_string("data_path","/home/xiaozhi/datasets/drugs/clean_17p_v1_splited/","cleaned data folder path")
```
Comment (shengwa) [inline regression_logp.py]: Please point to a common space or leave blank, if leave blank, please give an example usage of this.

File: regression_logp.py [hunk 12]
```diff
@@ -0,0 +1,168 @@
...
+flags.DEFINE_integer("i",0,"from 0 to 16")
```
Comment (shengwa) [inline regression_logp.py]: Check format, or format yourself.

File: neuralfingerprint/io_utils.py [hunk 13]
```diff
@@ -3,36 +3,35 @@
...
-    return [(data[0][s], data[1][s]) for s in slices]
+
+def load_data(filename, input_name, target_name):
+    return load_data_slices_nolist(filename, input_name, target_name)
+
+
+def load_data_slices_nolist(filename, input_name, target_name):
+    train_file = filename[0]
+    val_file = filename[1]
+    test_file = filename[2]
```
Comment (shengwa) [inline neuralfingerprint/io_utils.py:29]: Please make sure the filename contains three ones.

File: neuralfingerprint/io_utils.py [hunk 14]
```diff
@@ -3,36 +3,35 @@
...
+    data0 = read_csv(train_file, input_name, target_name)
```
Comment (shengwa) [inline neuralfingerprint/io_utils.py]: Please rename data0 1 2 to meaningful names.

File: regression_logp.py [hunk 15]
```diff
@@ -0,0 +1,177 @@
...
+from neuralfingerprint import load_data
+from neuralfingerprint import build_morgan_deep_net
+from neuralfingerprint import build_conv_deep_net
+from neuralfingerprint import normalize_array, adam
+from neuralfingerprint import build_batched_grad
+from neuralfingerprint.util import rmse
+
+from autograd import grad
+
+flags.DEFINE_string("data_path",
```
Comment (shengwa) [inline regression_logp.py]: See last comment!!!!!

File: regression_logp.py [hunk 16]
```diff
@@ -0,0 +1,177 @@
...
+                    "/home/xiaozhi/datasets/drugs/clean_17p_v1_splited/",
+                    "cleaned data folder path")
+flags.DEFINE_integer("i", 0, "from 0 to 16")
+FLAGS = flags.FLAGS
+
+data_path = FLAGS.data_path
+pi = FLAGS.i
```
Comment (shengwa) [inline regression_logp.py]: `p_i`?

File: README.md [hunk 17]
```diff
@@ -20,6 +20,7 @@ This package requires:
 * Scipy version >= 0.15.0  (For Anaconda users, use `conda install scipy` instead of `pip install scipy` to upgrade your version.)
 * [RDkit](http://www.rdkit.org/docs/Install.html)
 * [Autograd](http:github.com/HIPS/autograd) (Just run `pip install autograd`)
+* [Smile](https://github.com/uta-smile/smile-python) (Just run `pip install smile`)
 
 ## Examples
```
Comment (shengwa) [inline README.md:30]: Can you also give an example how to run `regression.py` ?

File: regression_logp.py [hunk 18]
```diff
@@ -0,0 +1,177 @@
+# Example regression script using neural fingerprints.
```
Comment (shengwa) [inline regression_logp.py]: Rename `regression_logp.py` to `regression.py`

**Review by shengwa — COMMENTED** [review]
Almost there!

File: neuralfingerprint/io_utils.py [hunk 19]
```diff
@@ -15,18 +15,38 @@ def read_csv(filename, input_name, target_name):
 
 
 def load_data(filename, input_name, target_name):
-    return load_data_slices_nolist(filename, input_name, target_name)
+    if len(filename) == 3:
+        return load_data_slices_islist(filename, input_name, target_name)
+    else:
+        raise Exception("Please make sure the filename contains three ones.")
+    # slices = []
```
Comment (shengwa) [inline neuralfingerprint/io_utils.py]: Remove??

File: README.md [hunk 20]
```diff
@@ -22,6 +22,11 @@ This package requires:
 * [Autograd](http:github.com/HIPS/autograd) (Just run `pip install autograd`)
 * [Smile](https://github.com/uta-smile/smile-python) (Just run `pip install smile`)
 
+## How to run
+* `python regression.py data_path='/smile/nfs/hm/17properties_datasets/17p_v1/single_property_xiaozhi/clean_17p_v1_splited' i=0`
+* data_path: Make sure there are three subfolders under the target data_path named train, val and test. if you mounted the nfs, please try this data_path: `/smile/nfs/hm/17properties_datasets/17p_v1/single_property_xiaozhi/clean_17p_v1_splited`
+* i: in this case, i is the number of 17 properties (pgp,solub,logp,2c9,2d6,2c19,rlm,mmp,ames,pldc,heptox,3a4,pampac,pampar,herg,1a2,vd), from 0 to 16
```
Comment (shengwa) [inline README.md]: add space after each property.

**Review by shengwa — APPROVED** [review]

**Pull request merged**